src/Router/Route.php

```diff
@@ -133,6 +151,14 @@ public function setOptions(array $options)
     }
 
     /**
+     * @param string $name
+     */
+    public function setName($name)
```

Comment:
@ezimuel The point is that if the name is changed *after* the route has been added to the underlying router, the change will not have the expected outcome. As an example:

```php
$app->post('/foo', 'Foo::bar')->setName('foo-bar');
```

implies that I can reference the route by the name `foo-bar` when working with the underlying router implementation, but, in fact, I cannot.

Technically, this is true for every other aspect encapsulated by `Route` as well, including `options`, as those are set as soon as `addRoute()` is called on the routing adapter. The only way having mutable properties makes sense is if we delay passing the routes to the underlying router until we actually are ready to route.